services/mailservice/inbucket.go

```diff
@@ -102,17 +101,20 @@ func GetMessageFromMailbox(email, id string) (results JSONMessageInbucket, err e
 	}
 	defer emailResponse.Body.Close()
 
-	err = json.NewDecoder(emailResponse.Body).Decode(&record)
+	if err = json.NewDecoder(emailResponse.Body).Decode(&record); err != nil {
+		return record, err
```

Comment:
Can we `return nil, err` here and on line 100? Unsure if we should do it throughout the method or not, but I believe it's idiomatic to not return a partial result when there's an error return value.

Reply:
The return value of `GetMessageFromMailbox()` is `(JSONMessageInbucket, error)`, not `(*JSONMessageInbucket, error)`. So I don't think there is a way to return to `return nil, err`.

Site note: It seams like this function is only used in test in TE. Can you check EE please? Maybe this is dead code.

Reply:
It isn't used in `enterprise` code, though I suspect it's the backbone of the tests in question so as to verify some functionality. Thanks for clarifying re: return values -- I'd missed that it's not a pointer type!